### PR TITLE
Remote Control - Seat

### PR DESCRIPTION
**Pull request by yang1070**
SDL remote control allows a user to control certain modules of the vehicle. This proposal allows mobile applications control powered seats within a vehicle (while the vehicle is in parking). A mobile application can send a remote control command to adjust seat positions, turn on or off cooled/heated seat, or change massage settings. The application can also get the status of each seat as well as receive notifications once the status of any seat changes.

### CODE REVIEW

**Comment by yang1070** [comment]
@theresalech this proposal is ready for review. Thanks.

File: proposals/nnnn-remote-control-seat.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+Add a `SEAT` RC module, the capabilities of each seat, and the related control data type to `Mobile_API.xml` and `HMI_API.xml`
+
+| Control Item | Value Range |Type | Comments |
+| ------------ |------------ | ------------ | ------------ |
+| Seat Heating Enabled | true, false | Get/Set/Notification | Indicates whether heating is enabled for a seat |
+| Seat Cooling Enabled | true, false | Get/Set/Notification | Indicates whether cooling is enabled for a seat |
+| Seat Heating  level | 0-100% | Get/Set/Notification | Level of the seat heating |
+| Seat Cooling  level | 0-100% | Get/Set/Notification | Level of the seat cooling |
+| Seat Horizontal Positon | 0-100% | Get/Set/Notification | Adjust a seat forward/backward, 0 means the nearest position to the steering wheel, 100% means the furthest position from the steering wheel |
+| Seat Vertical Position | 0-100% | Get/Set/Notification | Adjust seat height (up or down) in case there is only one actutator for seat height, 0 means the lowest position, 100% means the highest position|
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please change "actutator" to "actuator"

File: proposals/nnnn-remote-control-seat.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+| Seat-Front Vertical Position | 0-100% | Get/Set/Notification | Adjust seat front height (in case there are two actutators for seat height), 0 means the lowest position, 100% means the highest position|
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please change "actutators" to "actuators"

File: proposals/nnnn-remote-control-seat.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+| Seat-Back Vertical Position | 0-100% | Get/Set/Notification | Adjust seat back height (in case there are two actutators for seat height), 0 means the lowest position, 100% means the highest position|
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please change "actutators" to "actuators"

File: proposals/nnnn-remote-control-seat.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  <description>List possible modes of a massage zone.</description>
+  <element name="OFF"/>
+  <element name="LOW"/>
+  <element name="HIGH"/>
+</enum>
+
+<struct name="MassageModeData">
+  <description>Specify the mode of a massage zone.</description>
+  <param name="massageZone" type="MassageZone" mandatory="true"></param>
+  <param name="massagegMode" type="MassageMode" mandatory="true"></param>
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please change "massagegMode" to "massageMode"

File: proposals/nnnn-remote-control-seat.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+  </param>
+  <param name="headSupportHorizontalPositionAvailable" type="Boolean" mandatory="false">
+  </param> 
+  <param name="headSupportVerticalPositionAvailable" type="Boolean" mandatory="false">
+  </param> 
+  <param name="massageEnabledAvailable" type="Boolean" mandatory="false">
+  </param> 
+  <param name="massageModeAvailable" type="Boolean" mandatory="false">
+  </param> 
+  <param name="massageCushionFirmnessAvailable" type="Boolean" array="false">
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Should 'array="false"' be 'mandatory="false"'?

File: proposals/nnnn-remote-control-seat.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+  </param>
+  <param name="memoryAvailable" type="Boolean" array="false">
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Should 'array="false"' be 'mandatory="false"'?

File: proposals/nnnn-remote-control-seat.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
+# Remote Control - Seat
+
+* Proposal: [SDL-NNNN](nnnn-remote-control-seat.md)
+* Author: [Zhimin Yang](https://github.com/yang1070)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Android / RPC ]
+
+## Introduction
+
+SDL remote control allows a user to control certain modules of the vehicle. This proposal allows mobile applications control powered seats within a vehicle (while the vehicle is in parking). A mobile application can send a remote control command to adjust seat positions, turn on or off cooled/heated seat, or change massage settings. The application can also get the status of each seat as well as receive notifications once the status of any seat changes.
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please update "This proposal allows mobile applications control powered seats..." to "This proposal allows mobile applications to control powered seats..."

File: proposals/nnnn-remote-control-seat.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+  <param name="seatControlData" type="SeatControlData" mandatory="false">
+  </param>
+</struct>
+```
+#### Additions to HMI_API
+The changes are the same as mobile api changes, they are not duplicated here. 
+
+## Potential downsides
+
+Due to the lack of an agreed control resource ID/Zone scheme, we define `SupportedSeat` enum as a work around to allow control both driver seat and front passenger seat.
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please update "...to allow control both driver seat..." to "...to allow control of both driver seat..."

File: proposals/nnnn-remote-control-seat.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+- New remote control modules support
+
+Mobile Apps:
+
+- There is no impact on existing mobile applications or SDL integrations if they do not used the new module or new parameters.
+
+Policy:
+
+- Policies need to support new ModuleTypes (`SEAT`)
+- SDL need to transfer RPCs with new parameters to appropriate vehicle's module. The response resultCode depends on vehicle's result of processing.
```
Comment (theresalech) [inline proposals/nnnn-remote-control-seat.md]: Please update to "SDL needs to transfer RPCs with new parameters to appropriate vehicle modules."

File: proposals/nnnn-remote-control-seat.md (hunk 7 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: updated, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 1 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: updated, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 2 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: updated, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 3 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: updated, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 4 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: changed, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 5 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: you are correct, changed, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 6 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: changed, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 9 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: changed, thank you

File: proposals/nnnn-remote-control-seat.md (hunk 8 above)
Comment (yang1070) [reply proposals/nnnn-remote-control-seat.md]: changed,  thank you

**Comment by yang1070** [comment]
@theresalech Thank you very much for your review. I have updated the proposal according to your comments.

**Pull request merged**